backend/src/apiserver/server/run_server.go

```diff
@@ -72,23 +72,35 @@ func (s *RunServer) ListRuns(ctx context.Context, request *api.ListRunsRequest)
 }
 
 func (s *RunServer) ArchiveRun(ctx context.Context, request *api.ArchiveRunRequest) (*empty.Empty, error) {
-	err := s.resourceManager.ArchiveRun(request.Id)
+	err := CanAccessRun(s.resourceManager, ctx, request.Id)
```

Comment:
CanAccessRun seems run specific. could you move to this file instead and mark it private, for readability?

Reply:
and if you make this function part of RunServer, you could skip passing in resource manager.

Reply:
make sense.
Done